### PR TITLE
[misc] Implement cache file cleaning

### PR DESCRIPTION
**Pull request by PGZXB**
Related issue = #4401

### CODE REVIEW

File: taichi/runtime/llvm/llvm_offline_cache.h [hunk 1]
```diff
@@ -129,7 +144,21 @@ class LlvmOfflineCacheFileReader {
 };
 
 class LlvmOfflineCacheFileWriter {
+  enum CleanCacheFlags {
+    NotClean = 0b000,
+    CleanOldVersion = 0b001,
+    CleanOldUsed = 0b010,
+    ClaenOldCreated = 0b100
```
Comment (lin-hitonami) [inline taichi/runtime/llvm/llvm_offline_cache.h:151]: Typo: Clean

File: taichi/runtime/llvm/llvm_offline_cache.h [hunk 2]
```diff
@@ -129,7 +144,21 @@ class LlvmOfflineCacheFileReader {
...
+  };
+
  public:
+  enum CleanCachePolicy {
+    Nerver = NotClean,
```
Comment (lin-hitonami) [inline taichi/runtime/llvm/llvm_offline_cache.h:156]: Typo: Never

File: taichi/runtime/llvm/llvm_offline_cache.cpp [hunk 3]
```diff
@@ -20,6 +31,33 @@ namespace {
 using Format = LlvmOfflineCache::Format;
 constexpr char kMetadataFilename[] = "metadata";
 
+static bool check_llvm_cache_verison(const LlvmOfflineCache::Version &ver) {
```
Comment (lin-hitonami) [inline taichi/runtime/llvm/llvm_offline_cache.cpp:34]: I suggest changing the function name to `is_current_llvm_cache_version`

File: taichi/runtime/llvm/llvm_offline_cache.h (hunk 1 above)
Comment (PGZXB) [reply taichi/runtime/llvm/llvm_offline_cache.h:151]: Thanks

**Review by lin-hitonami — APPROVED** [review]
LGTM

**Pull request merged**